File: 2026-03/src/day_17/boj_1439/Main.java
package day_17.boj_1439;

import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        String s = br.readLine();
        int cnt0 = 0;
        int cnt1 = 0;
        char prv = '\u0000';

        for (char c : s.toCharArray()) {
            if (c != prv) {
                if (c == '0') {
                    cnt0++;
                } else {
                    cnt1++;
                }
            }

            prv = c;
        }

        System.out.println(Math.min(cnt0, cnt1));
    }
}

File: 2026-03/src/day_17/boj_1439/Main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);

    string s;
    cin >> s;

    int cnt0 = 0;
    int cnt1 = 0;
    char prv = '\0';

    for (char c : s) {
        if (c != prv) {
            if (c == '0') {
                cnt0++;
            } else {
                cnt1++;
            }
        }

        prv = c;
    }

    cout << min(cnt0, cnt1);
}
